packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -117,30 +93,100 @@ library LibProposing {
+        // of multiple Taiko blocks being proposed within a single
+        // Ethereum block, we must introduce a salt to this random
+        // number as the L2 mixHash.
+        unchecked {
+            meta.difficulty = meta.blobHash
+                ^ bytes32(block.prevrandao * b.numBlocks * block.number);
+        }
+
+        // Use the difficulty as a random number
+        meta.minTier = ITierProvider(resolver.resolve("tier_provider", false))
+            .getMinTier(uint256(meta.difficulty));
+
+        // Update certain meta fields
+        if (txList.length > 0) {
+            // The proposer must be an Externally Owned Account (EOA) for
+            // calldata usage. This ensures that the transaction is not an
+            // internal one, making calldata retrieval more straightforward.
+            if (!LibAddress.isSenderEOA()) revert L1_PROPOSER_NOT_EOA();
+
+            if (params.txListByteOffset != 0 || params.txListByteSize != 0) {
+                revert L1_TXLIST_OFFSET_SIZE();
+            }
+
+            // blockMaxTxListBytes is a uint24
+            if (txList.length > config.blockMaxTxListBytes) {
+                revert L1_TXLIST_TOO_LARGE();
+            }
+
+            meta.blobHash = keccak256(txList);
+            meta.txListByteOffset = 0;
+            meta.txListByteSize = uint24(txList.length);
+        } else {
+            if (!config.blobAllowedForDA) {
+                revert L1_BLOB_FOR_DA_DISABLED();
+            }
+
+            if (params.blobHash != 0) {
+                // We try to reuse an old blob
+                if (isBlobReusable(state, config, params.blobHash)) {
+                    revert L1_BLOB_NOT_REUSEABLE();
+                }
+                meta.blobHash = params.blobHash;
+            } else {
+                // Always use the first blob in this transaction. If the
+                // proposeBlock functions are called more than once in the same
+                // L1 transaction, these multiple L2 blocks will share the same
+                // blob.
+                meta.blobHash = IBlobHashReader(
+                    resolver.resolve("blob_hash_reader", false)
+                ).getFirstBlobHash();
+
+                if (meta.blobHash == 0) revert L1_NO_BLOB_FOUND();
+
+                // Depends on the blob data price, it may not make sense to
+                // cache the blob which costs 20,000 (sstore) + 631 (event)
+                // extra gas.
+                if (params.cacheBlobForReuse) {
+                    state.reusableBlobs[meta.blobHash] = block.timestamp;
+                    emit BlobCached(meta.blobHash);
+                }
+            }
+
+            if (params.txListByteSize == 0) revert L1_TXLIST_OFFSET_SIZE();
+            if (
+                uint256(params.txListByteOffset) + params.txListByteSize
+                    > config.blockMaxTxListBytes
+            ) {
+                revert L1_TXLIST_TOO_LARGE();
+            }
```

Comment:
I think this check is only useful if `config.blockMaxTxListBytes` is set to the blob size, because if it's not the blob size we still have to check in the circuit if the range is valid anyway. But the same constant is also used with the calldata tx list so not sure if that will be the case or not. Currently they would have very similar limits, so setting it to the blob size would make sense.

Reply:
Change the code to also check against MAX_BYTES_PER_BLOB